### ISSUE TITLE
Set a sane default for the list save location

### ISSUE THREAD

**Issue by Andy2244** [vosmiic/jellyfin-ani-sync#21]
`jellyfin_ani_sync.TaskUpdateAnimeList: User has not set anime list save location; skipping`

There should be a safe location we can use as default save location, which is also pre-filled in the settings field. A normal user may not understand how crucial this list is.

I also was under the assumption that via default/empty setting the plugin would actually use the plugin folder or some default Jellyfin temp/lib folder to store this list.

I would rephrase this `This is not necessary, but will make anime detection much more reliable`, to reflect that only with AniList as metadata provider the list is not needed, otherwise this should be mandatory given all the lookup issues.

There should also be a hint/advise to only use one metadata provider per anime library, based on what folder/file naming is used, otherwise just stacking multiple AniList/AniDB/TvDB/TMDB will lead to mixed results. I would assume most users are not aware of the issue that "normal" Anime has no official seasons and the TvDB/TMDB season ordering/numbering is kinda arbitrary and manually maintained.

_Main point is a normal user may not check the server logs and just assumes this plugin works like trakt with 100% accuracy, without understanding the inherent problems regarding finding the correct Anime/season/episode, depending on the metadata source._

**Comment by Andy2244** [comment]
I also can not set the path even manually?

If i try `cache\ani-sync` i get `Error: Could not find a part of the path 'D:\jellyfin\cache\ani-sync\test\eyjak5cp.3ua'` ?

I also think `cache\ani-sync` should be the default, since thats where all the other plugins seems to store its data.

**Comment by vosmiic** [comment]
>There should be a safe location we can use as default save location

Agreed.

>I think cache\ani-sync should be the default

Sounds good, don't suppose you could share which plugins store their data there? I have had a lookup at some other plugins and they seem to either ask for the users input or store them in the users library locations.
>otherwise this should be mandatory given all the lookup issues

Hmm maybe I could word it so that it is heavily suggested, but I have been using the plugin before the changes and it would find most of the shows and correctly update them (I understand others may have had other experiences). Don't get me wrong, the solution that you came up with has now taken successful anime lookup from about 65% to 99% (basically the only time it won't identify an anime is if it isn't contained in the list, and the fallback method doesn't work) so its a massive help, but I wouldn't call it mandatory. That would suggest the plugin doesn't work without it.

>There should also be a hint/advise to only use one metadata provider per anime library, based on what folder/file naming is used, otherwise just stacking multiple AniList/AniDB/TvDB/TMDB will lead to mixed results.

In theory it should work no matter how many metadata providers you have provided, as internally there is a priority system of
1. AniList
2. AniDB
3. TvDB
4. Others

So if you have an AniList external ID attached to the show you are watching, it will always pick that over and above any other IDs.
And the season/episode offset shouldn't (in theory) be a problem as the anime list XML file comes with episode offsets which are used by the plugin. 
For example, if you are watching a show that has multiple cours where the first cour ends at episode 12, and the plugin detects you are watching episode 13 (because the metadata provider you are using doesn't support cours, including TvDB), it will see that the first season ends at 12 and so presume that you are watching the first episode of the 2nd cour. 

That being said I will add a suggestion to stick to a single metadata provider (in the above suggested priority) as it makes everything much cleaner.

**Comment by Andy2244** [comment]
> Hmm maybe I could word it so that it is heavily suggested

Yeah thats what i meant, it mainly depends on what you test the plugin with.
In my tests i used mainly continuing seasons in tvdb format, so i had many wrong or missing lookup's.


> Sounds good, don't suppose you could share which plugins store their data there?
I see those dirs at least: `anilist, omdb, temp`

**Comment by Andy2244** [comment]
Ah btw verify that the new default save location works correctly with the `--datadir` option, since that's often used, especially with docker containers. I assume a plugin can ask for the datadir location?

**Comment by Andy2244** [comment]
Just noticed the "cache dir" is a actual server setting under generals and can be changed.
It states `Specify a custom location for server cache files such as images. Leave blank to use the server default.`

So try grab this settings field or default and create a `ani-sync` subfolder in it.

**Comment by vosmiic** [comment]
The default anime save list location (at least for Linux, in Windows it will be different) is now: `/home/{{User}}/.cache/jellyfin/anisync/` or whatever the server cache is (its dynamic).

As always, this is available now in master, but will be available in the next release.
